### PR TITLE
feat(argo-cd): Add sizeLimit parameters on emptyDir

### LINKED ISSUE #2394: Add Helm Chart Values for Configuring Requests and Limits on EmptyDir for ArgoCD
### Is your feature request related to a problem?

I am currently deploying ArgoCD on Kubernetes using Helm, and I have integrated Kyverno to enforce security best practices. However, I have encountered an issue where Kyverno is blocking my ArgoCD deployment due to the absence of resource requests and limits on the EmptyDir volumes used by ArgoCD components like Deployment and StatefulSet.

The Helm chart for ArgoCD currently lacks the flexibility to configure these resource settings, making it challenging for users like me who rely on Kyverno for policy enforcement.

### Related helm chart

argo-cd

### Describe the solution you'd like

I propose enhancing the ArgoCD Helm chart by adding new configurable values to allow users to set resource requests and limits for EmptyDir volumes. This addition would provide users with the flexibility to customize these settings based on their specific deployment requirements.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pierreblais**
Resolve #2394 

Checklist:

* [X] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [X] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [X] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -26,5 +26,5 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
-    - kind: changed
-      description: Bump argo-cd to v2.10.4
+    - kind: feat
```
Comment (tico24) [inline charts/argo-cd/Chart.yaml:29]: this should be `kind: added`

**Comment by pierreblais** [comment]
I changed It ! Anything else ?

**Comment by tico24** [comment]
> I changed It ! Anything else ?

You should test locally and make the tests pass:
```
Error: INSTALLATION FAILED: Deployment in version "v1" cannot be handled as a Deployment: json: cannot unmarshal string into Go struct field Volume.spec.template.spec.volumes.emptyDir of type v1.EmptyDirVolumeSource
```

**Comment by pierreblais** [comment]
Alright, it should be good !

**Comment by yu-croco** [comment]
@pierreblais 
`version` in Chart.yaml is left behind. Please bump it.

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 2]
```diff
@@ -334,7 +334,13 @@ spec:
         {{- toYaml . | nindent 6 }}
       {{- end }}
       - name: argocd-home
+        {{- if .Values.controller.emptyDir.sizeLimit }}
+        emptyDir:
+          sizeLimit: {{ .Values.controller.emptyDir.sizeLimit }}
+        {{- else }}
         emptyDir: {}
+        {{- end }}
+
```
Comment (tico24) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:343]: Nit: line break not needed here

Comment (pierreblais) [reply charts/argo-cd/templates/argocd-application-controller/deployment.yaml:343]: I'll fixed

**Comment by pierreblais** [comment]
I think we are ok ?

**Comment by yu-croco** [comment]
> I think we are ok ?

Hi @pierreblais , please fix below.

- bump version in Chart.yaml
- fix failed CI
- resolve conflict

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 3]
```diff
@@ -354,4 +360,4 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       dnsPolicy: {{ .Values.controller.dnsPolicy }}
-{{- end }}
+{{- end }}
```
Comment (tico24) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:363]: nit: missing newline

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM.

**Review by tico24 — APPROVED** [review]

**Pull request merged**